vite.plugins.ts

```diff
@@ -0,0 +1,94 @@
+import { render } from "@lit-labs/ssr/lib/render-lit-html";
+import { readFileSync } from "fs";
+import { PluginOption } from "vite";
+import { pageContent as aboutStaticContent } from "./src/frontend/src/flows/about";
+import { I18n } from "./src/frontend/src/i18n";
+
+/**
+ * Read the II canister ID from dfx's local state
+ */
+const readCanisterId = (): string => {
+  const canisterIdsJsonFile = "./.dfx/local/canister_ids.json";
+
+  try {
+    const {
+      internet_identity: { local: canisterId },
+    } = JSON.parse(readFileSync(canisterIdsJsonFile, "utf-8"));
+
+    const assertNonNullish: (
+      value: string
+    ) => asserts value is NonNullable<string> = (value: string): void => {
+      if (value === null || value === undefined) {
+        throw new Error("Internet identity canister ID undefined");
+      }
+    };
+
+    assertNonNullish(canisterId);
+
+    console.log("Read canister ID:", canisterId);
+
+    return canisterId;
+  } catch (e) {
+    throw Error(`Could not get canister ID from ${canisterIdsJsonFile}: ${e}`);
+  }
+};
+
+/**
+ * Inject the II canister ID as a <script /> tag in index.html for local development.
+ */
+export const injectCanisterIdPlugin = (): {
+  name: "html-transform";
+  transformIndexHtml(html: string): string;
+} => ({
+  name: "html-transform",
+  transformIndexHtml(html): string {
+    return html.replace(
+      `<script id="setupJs"></script>`,
+      `<script data-canister-id="${readCanisterId()}" id="setupJs"></script>`
+    );
+  },
+});
+
+/**
+ * Remove the <script /> that loads the index.js for production build.
+ * A script loader is injected by the backend.
```

Comment:
Ah that's a shame, is there no vite option for not injecting the js?

Reply:
Don't know. Reused the work I did few months ago. I guess I had a look at that time.